feature(esp_tinyusb): Added config option to select USB OTG1.1 periph on P4

roma-jam opened the pull request:
## Description

Added possibility to select the USB OTG1.1 Peripheral (FS) on ESP32P4 via menuconfig.

## Related
N/A

## Testing
Could be tested only on correct P4 board. 

## Limitation
- This changes are available only after removing `usb_phy_p4.c` in esp-idf. Removing the specific P4-related file additionally unlocks the possibility to initialize PHY for USB OTG1.1, which it necessary for USB OTG1.1 Peripheral.  

## Checklist

Before submitting a Pull Request, please ensure the following:

- [x] 🚨 This PR does not introduce breaking changes.
- [x] All CI checks (GH Actions) pass.
- [x] Documentation is updated as needed.
- [x] Tests are updated or added as necessary.
- [x] Code is well-commented, especially in complex areas.
- [x] Git history is clean — commits are squashed to the minimum necessary.

tore-espressif commented on device/esp_tinyusb/Kconfig:
```diff
@@ -11,10 +11,14 @@ menu "TinyUSB Stack"
         default TINYUSB_RHPORT_HS
         help
-            Allows set the USB PHY Controller for TinyUSB: HS (USB OTG2.0 PHY for HighSpeed)
+            Allows set the USB PHY Controller for TinyUSB:
+            - HS (USB OTG2.0 PHY for HighSpeed)
+            - FS (USB OTG1.1 PHY for FullSpeed)
 
         config TINYUSB_RHPORT_HS
             bool "HS"
+        config TINYUSB_RHPORT_FS
+            bool "FS"
```
If possible, I'd like the peripheral selection to be runtime configurable from the very first release.

Could you please investigate this option more? IIRC, it is possible only from our tinyusb v0.17.0~1 onwards

roma-jam replied:
Yeah, sure.
This PR just was already there, so I thought we could use it, provide the configurable option and then change it to dynamically configurable feature. I’ll check.

tore-espressif replied:
BTW, this will not work without the PHY changes in esp-idf, right? 

If we need the esp-idf changes, then we can postpone this feature to v1.7.0.... Up to you :)

tore-espressif reviewed (approved):
@roma-jam Please add a note to changelog.md and make sure to emphasize that the users need specific commit from esp-idf for this to work. Rest LGTM

roma-jam commented:
@tore-espressif I just revisit the configuration again, and I think that it might need some refactoring. 
Before it was: 
```
            Allows set the USB PHY Controller for TinyUSB:
            - HS (USB OTG2.0 PHY for HighSpeed)
            - FS (USB OTG1.1 PHY for FullSpeed)
```

Which are .... wrong. That is USB Peripheral, not PHY. So I changed it.

Also, we are not allowed to select PHY, it is done automatically, that is why I don't think there should a any mentions of "PHY"-word in the TinyUSB config. So I removed all of them and changed them with "Peripheral".

Also, I have added the option to enable-disable run-time peripheral port selection to proceed with dynamic configuration and keep the possibility to use the legacy way of port selection (just in case, to verify the hardware, because it is working).

tore-espressif commented on device/esp_tinyusb/tinyusb.c:
```diff
@@ -32,6 +32,13 @@ esp_err_t tinyusb_driver_install(const tinyusb_config_t *config)
         .otg_mode = USB_OTG_MODE_DEVICE,
+#if (USB_PHY_SUPPORTS_P4_OTG11)
+        .otg_speed = (TUD_OPT_HIGH_SPEED) ? USB_PHY_SPEED_HIGH : USB_PHY_SPEED_FULL,
+#else
+#if (CONFIG_IDF_TARGET_ESP32P4 && CONFIG_TINYUSB_RHPORT_FS)
+#error "USB PHY for OTG1.1 is not supported"
```
Or might be better:
`#error "USB PHY for OTG1.1 is not supported in this version of esp-idf. Please update your esp-idf."`

roma-jam replied:
@peter-marcisovsky No, not really. 

But it might be removed, when: 
- All versions of esp-idf, with P4 support will have USB PHY changes (means we finish the backporting)
- We will switch the configuration from build-time to run-time. 

Otherwise, there will be a possibility to use this version of esp_tinyusb with eps-idf without PHY support for the peripheral. :disappointed:

roma-jam replied:
Changed to: `USB PHY for OTG1.1 is not supported, please update your esp-idf.`

tore-espressif commented on device/esp_tinyusb/CHANGELOG.md:
```diff
@@ -1,6 +1,7 @@
 ## [Unreleased]
 
 - esp_tinyusb: Added possibility to configure NCM Transfer Blocks (NTB) via menuconfig
```
Not related to your changes but:
```suggestion
- NCM: Added possibility to configure NCM Transfer Blocks (NTB) via menuconfig
```

we can use `esp_tinyusb` category for changes outside of `tinyusb`

roma-jam replied:
That comes from here: https://github.com/espressif/esp-usb/pull/114

I will refactor it during the very specific PR, pushing the release of component. 
Anyway, we need to change `## [Unreleased]` there.

roma-jam replied:
Changes took place in here: https://github.com/espressif/esp-usb/pull/120
